.github/workflows/tests-e2e-upgrade.yaml

```diff
@@ -636,6 +632,15 @@ jobs:
           kubectl get pods --all-namespaces -o wide
           kubectl -n kube-system exec daemonset/cilium -- cilium status
 
+      - name: Restart LRP backend pod
+        if: ${{ matrix.skip-upgrade != 'true' && matrix.local-redirect-policy == 'true' }}
+        shell: bash
+        run: |
+          # Restart lrp-backend pod to set the netns cookie to the endpoint
+          testns=$(kubectl get deploy --all-namespaces -o jsonpath="{.items[?(@.metadata.name=='lrp-backend')].metadata.namespace}") \
+            && kubectl rollout restart -n $testns deploy/lrp-backend \
+            && kubectl rollout status -n $testns deploy/lrp-backend
+
```

Comment:
Nice! Strictly this would only be needed when using SocketLB, correct? (not saying we should add such a fine-granular check here)

I believe this special-handling is essentially what we mention in the docs [here](https://docs.cilium.io/en/latest/network/kubernetes/local-redirect-policy/#advanced-configurations) ? Maybe it's worth fine-tuning the docs to say that it only applies to SocketLB?

Reply:
Actually the per packet LB also relies on the same netns cookie. So this handling is needed for the per packet LB too. 

https://github.com/cilium/cilium/blob/v1.16/bpf/bpf_lxc.c#L119
https://github.com/cilium/cilium/blob/v1.16/pkg/datapath/loader/template.go#L271

Reply:
That's odd - why didn't the configs with per-packet LB fail during the upgrade test? :thinking:

Reply:
That's because we run `local-redirect-policy` test on v1.16 only with SocketLB not to break the e2e-upgrade on main branch. 
https://github.com/cilium/cilium/blob/main/cilium-cli/connectivity/builder/local_redirect_policy.go#L28-L35

On main branch, we downgrade to v1.16.2, which doesn't have the per packet LB fix for LRP, and run connectivity test. So now we skip the per packet LB on v1.16. I will update this condition when the new v1.16 patch release comes out.

Reply:
:bulb: so the per-packet LB configs would have then started to fail with a newer CLI version. Thank you!